src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -284,6 +286,119 @@ public bool PrefixReserved
+            set
+            {
+                if (_packageDeprecationReasons != value)
+                {
+                    _packageDeprecationReasons = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasons));
+                }
+            }
+        }
+
+        private string _packageDeprecationReasonsExplained;
+        public string PackageDeprecationReasonsExplained
+        {
+            get { return _packageDeprecationReasonsExplained; }
+            set
+            {
+                if (_packageDeprecationReasonsExplained != value)
+                {
+                    _packageDeprecationReasonsExplained = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasonsExplained));
+                }
+            }
+        }
+
+        private string _packageDeprecationCustomMessage;
+        public string PackageDeprecationCustomMessage
+        {
+            get { return _packageDeprecationCustomMessage; }
+            set
+            {
+                if (_packageDeprecationCustomMessage != value)
+                {
+                    _packageDeprecationCustomMessage = value;
+                    OnPropertyChanged(nameof(PackageDeprecationCustomMessage));
+                }
+            }
+        }
+
+        private string _packageDeprecationAlternatePackageLinkText;
+        public string PackageDeprecationAlternatePackageLinkText
+        {
+            get { return _packageDeprecationAlternatePackageLinkText; }
+            set
+            {
+                if (_packageDeprecationAlternatePackageLinkText != value)
+                {
+                    _packageDeprecationAlternatePackageLinkText = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationAlternatePackageLinkText));
+                }
+            }
+        }
+
+        public string ExplainPackageDeprecationReasons(IEnumerable<string> reasons)
+        {
+            if (reasons.Contains(PackageDeprecationReason.CriticalBugs, StringComparer.OrdinalIgnoreCase))
+            {
+                if (reasons.Contains(PackageDeprecationReason.Legacy, StringComparer.OrdinalIgnoreCase))
+                {
+                    return "This package has been deprecated because it's legacy and has critical bugs.";
+                }
+                else
+                {
+                    return "This package has been deprecated because it has critical bugs.";
+                }
+            }
+            else if (reasons.Contains(PackageDeprecationReason.Legacy, StringComparer.OrdinalIgnoreCase))
+            {
+                return "This package has been deprecated because it's legacy and no longer maintained.";
+            }
+            else
+            {
+                return "This package has been deprecated.";
+            }
+        }
+
+        private static class PackageDeprecationReason
+        {
+            public const string CriticalBugs = "CriticalBugs";
```

Comment:
I think `CriticalBugs` should only be used as a enum-like value and not for display. `Critical bugs` or `Critical Bugs` is better for display

Reply:
Actually, this one is used as a enum-like value, and not for display ;)

Reply:
Oh okay. In your screenshot above it is "CriticalBugs" so I thought it was coming from here

Reply:
Yep, that's being served from the registration JSON blobs (as enum value). Will look into it.

The constants here are solely used to determine whether a particular reason is present to define the extended explanatory description.

Reply:
Addressed with https://github.com/NuGet/NuGet.Client/pull/2899/commits/f5c101bc81de17aa1060d51aa49b19bec14097cc